docs/csharp/programming-guide/classes-and-structs/snippets/local-functions/Program.cs

```diff
@@ -121,5 +121,27 @@ private async Task<int> SecondStep(int index, string name)
             await Task.Delay(100);
             return 9;
         }
+
+        //<YieldReturn>
+        public IEnumerable<string> SequenceToLowercase(IEnumerable<string> input)
+        {
+            if (input.Count() == 0)
+            {
+                throw new Exception("There are no items to convert to lowercase!");
```

Comment:
Throwing bare exceptions is not recommended. In this context it looks to be `ArgumentException`
```suggestion
                throw new ArgumentException("There are no items to convert to lowercase.");
```

Reply:
Normally I don't throw base exceptions, but my assumption was that the `yield return` was the focus of the demonstration. I'll replace it with `ArgumentException`